content/posts/blog/2020-10-30-consuming-public-content.md

```diff
@@ -0,0 +1,108 @@
+
+## What Are the Best Practices for Consuming Public Content
+
+Consuming public content, including open source projects & paid content, is key to maintaining the development velocity by which developers are required to serve their users. Which explains a Gartner study finding that [90% of software will include open source](https://www.cnet.com/news/90-percent-of-saas-providers-to-use-open-source-by-2010/). At the same time developers and consumers are pushing for velocity, our security and operations workers are pushing for a focus on the reliability and security of the public content.
+
+To balance the consumption with velocity desire against the reliability and security desires, we suggest adopting gated & verified workflows to enable the safe and secure use of rapidly deployed public content. While production deployments are the most obvious, maintaining a gated copy of the images you **build FROM** is just as important. Organizations should keep local copies of everything they depend upon, while building ci/cd workflows that update and validate their gated dependencies. For example, companies that maintained local gated copies of the javascript libraries they depend upon would have completely missed the production outages caused in the above cited example when the left-pad code was removed.
+
+## Short Term Throttling Mitigations
+
+Maintaining a gated and verified copy of the content you require will take time to implement. As we head into the end-of-year holiday period, organizations tend to lock down their development and deployments.
+
+When estimating the number of container image pull requests that may occur, take into account that when using cloud provider services or working behind a corporate NAT multiple users will be presented to Docker Hub in aggregate as a subset of IP addresses. This means your first anonymous request for a container image may be the 101st anonymous request from a specific shared IP address in the last six hours. Adding [Docker paid account authentication](https://docs.docker.com/docker-hub/download-rate-limit/#how-do-i-authenticate-pull-requests) to requests made to Docker Hub is a sure way to avoid potential service disruptions due to the new rate limit throttling that Docker is introducing on the first of November, 2020.
```

Comment:
```suggestion
When estimating the number of container image pull requests that may occur, take into account that when using cloud provider services or working behind a corporate NAT, multiple users will be presented to Docker Hub in aggregate as a subset of IP addresses. This means your first anonymous request for a container image may be the 101st anonymous request from a specific shared IP address in the last six hours. Adding [Docker paid account authentication](https://docs.docker.com/docker-hub/download-rate-limit/#how-do-i-authenticate-pull-requests) to requests made to Docker Hub is a sure way to avoid potential service disruptions due to the new rate limit throttling that Docker is introducing on the first of November, 2020.
```